src/modules/person/persistence/person.repository.ts

```diff
@@ -170,4 +243,32 @@ export class PersonRepository {
 
         return person;
     }
+    // In your PersonRepository class
+
+    public async deletePersonAndKontexte(
+        person: Person<true>,
+        permissions: PersonPermissions,
+    ): Promise<Result<void, DomainError>> {
+        // Check if the person has a keycloakUserId
+        if (!person.keycloakUserId) {
+            throw new PersonHasNoKeycloakId(person.id);
+        }
+        // Delete the person from Keyclock
+        await this.kcUserService.delete(person.keycloakUserId);
+        // First, delete all kontexte for the personId
+        const kontextResponse: Result<void, DomainError> =
+            await this.dBiamPersonenkontextRepo.deletePersonenkontexteByPersonId(person.id);
```

Comment:
Like this there is no transaction around the deletion of a person and her kontexts. That is not nice, but acceptable.
In this case it is fine to set `cascade: [Cascade.REMOVE]` in the person.entity.ts in the personenkontext's `@OneToMany`

Reply:
Does cascade: [Cascade.REMOVE] really helps with anything in our case? The personId is a foreign key in the entity personkontexte that references the persons entity "id" so you can't really delete a person if it has at least 1 kontext still available.